monero-sys/src/lib.rs

```diff
@@ -1570,7 +1616,121 @@ impl FfiWallet {
+    /// Get the address for the given account and subaddress index.
+    pub fn address_at(&self, account_index: u32, address_index: u32) -> monero::Address {
+        // Reuse the private `address` helper
+        self.address(account_index, address_index)
+    }
+
+    /// Get the number of subaddresses for a given account.
+    pub fn num_subaddresses(&self, account_index: u32) -> usize {
+        ffi::numSubaddresses(&self.inner, account_index) as usize
+    }
+
+    /// Get the label for a specific subaddress.
+    pub fn subaddress_label(&self, account_index: u32, address_index: u32) -> String {
+        ffi::getSubaddressLabel(&self.inner, account_index, address_index)
+            .context("Failed to get subaddress label: FFI call failed with exception")
+            .expect("getSubaddressLabel should not fail")
+            .to_string()
+    }
+
+    /// Compute subaddress summaries for a given account index.
+    fn subaddress_summaries_sync(&mut self, account_index: u32) -> Vec<SubaddressSummary> {
+        let history_ptr = self
+            .inner
+            .pinned()
+            .history()
+            .context("Failed to get transaction history: FFI call failed with exception");
+
+        let Ok(history_ptr) = history_ptr else {
+            tracing::error!(error=%history_ptr.unwrap_err(), "Failed to get transaction history, proceeding with empty history");
+            let size = self.num_subaddresses(account_index) as u32;
+            return (0..size)
+                .map(|idx| {
+                    let address = self.address_at(account_index, idx);
+                    let label = self.subaddress_label(account_index, idx);
+                    SubaddressSummary {
+                        account_index,
+                        address_index: idx,
+                        address,
+                        label,
+                        received: 0,
+                        tx_count: 0,
+                    }
+                })
+                .rev()
+                .collect();
+        };
+
+        let history = unsafe {
+            Pin::new_unchecked(
+                history_ptr
+                    .as_mut()
+                    .expect("history pointer to not be null after we just checked"),
+            )
+        };
+        let _ = history
+            .refresh()
+            .context("Failed to refresh transaction history: FFI call failed with exception")
+            .inspect_err(|e| tracing::error!(error=%e,"Failed to refresh transaction history"));
+
+        let history_handle = TransactionHistoryHandle(history_ptr);
+        let count = history_handle.count();
+
+        let size = self.num_subaddresses(account_index) as u32;
+        let mut received: Vec<u64> = vec![0; size as usize];
+        let mut tx_count: Vec<u32> = vec![0; size as usize];
+
+        for i in 0..count {
+            if let Some(tx_info) = history_handle.transaction(i) {
```

Comment:
Is there a good reason this can sometimes fail? Otherwise we should probably be explicit about the error and return an error (`anyhow::bail!()`)